test/groovy/MulticloudDeployTest.groovy

```diff
@@ -276,4 +271,52 @@ class MulticloudDeployTest extends BasePiperTest {
         assert cloudFoundryDeployRule.hasParameter('deployTool', 'cf_native')
     }
 
+    @Test
+    void 'cfCreateServices with parallelTestExecution defined in compatible parameter - must run in parallel'() {
+        def closureRun = null
+
+        helper.registerAllowedMethod('error', [String.class], { s->
+            if (s == "Deployment skipped because no targets defined!") {
+                // This error is ok because in this test we're not interested in the deployment
+            } else {
+                throw new RuntimeException("Unexpected error in test with message: ${s}")
+            }
+        })
+        helper.registerAllowedMethod('parallel', [Map.class], {m -> closureRun = m})
+
+        nullScript.commonPipelineEnvironment.configuration.general['features'] = [parallelTestExecution: true]
+
+        stepRule.step.multicloudDeploy([
+            script: nullScript,
+            cfCreateServices: [[serviceManifest: 'services-manifest.yml', space: 'PerformanceTests', org: 'foo', credentialsId: 'foo']],
+            source: 'file.mtar'
+        ])
+
+        assert closureRun != null
+    }
+
+    @Test
+    void 'cfCreateServices with parallelExecution defined - must run in parallel'() {
+        def closureRun = null
+
+        helper.registerAllowedMethod('error', [String.class], { s->
+            if (s == "Deployment skipped because no targets defined!") {
+                // This error is ok because in this test we're not interested in the deployment
+            } else {
+                throw new RuntimeException("Unexpected error in test with message: ${s}")
+            }
+        })
+        helper.registerAllowedMethod('parallel', [Map.class], {m -> closureRun = m})
+
+        nullScript.commonPipelineEnvironment.configuration.general = [parallelExecution: true]
+
+        stepRule.step.multicloudDeploy([
+            script: nullScript,
+            cfCreateServices: [[serviceManifest: 'services-manifest.yml', space: 'PerformanceTests', org: 'foo', credentialsId: 'foo']],
+            source: 'file.mtar'
+        ])
+
+        assert closureRun != null
```

Comment:
Instead of just asserting the parrellel execution, I am missing a test which checks for the service creation itself. E.g. that cfCreateService is called.

Reply:
dito -> `test/groovy/util/JenkinsMockStepRule.groovy`

Reply:
You're right, I've added an additional test case for that.